images/linux/scripts/base/apt-mock.sh

```diff
@@ -14,19 +16,44 @@ while [ \$i -le 30 ];do
   err=\$(mktemp)
   $real_tool "\$@" 2>\$err
   result=\$?
+  cat \$err >&2
   if [ \$result -eq  0 ];then
+    # no errors, continue
+    rm \$err
     break
   fi
+  # some error exists, check for apt db locked
   grep -q 'Could not get lock' \$err
-  held=\$?
-  if [ \$held -ne  0 ];then
-    break
+  result=\$?
+  if [ \$result -ne  0 ];then
```

Comment:
It think this code should be refactored before merging. 
Currently, It contains 4 nested if to check 4 grep and it is absolutely unreadable structure already. What if we add more commands in future?

I suggest using one of the following options:
```bash
grep -q 'Could not get lock' \$err && \
grep -q 'Could not open file /var/lib/apt/lists' \$err && \
grep -q 'IPC connect call failed' \$err && \
held=\$?
if [ \$held -ne  0 ];then
  break
fi
```

Or even single grep with multipattern option:
```bash
optionA="Could not get lock"
optionB="Could not open file /var/lib/apt/lists"
optionC="IPC connect call failed"
grep -q '$optionA\|$optionB\|$optionC' \$err
held=\$?
if [ \$held -ne  0 ];then
  break
fi
```

Reply:
yeah, i refactored the code, but in a bit different way. The code your suggested assumes there is the same handler 
for every error case that is not generally true. For example, currently there's no evidence the IPC problem could be solved with a delay.

Reply:
Thank you, I like the your approach!